kotlin/workflow-testing/src/test/java/com/squareup/workflow/testing/RenderTesterTest.kt

```diff
@@ -0,0 +1,144 @@
+package com.squareup.workflow.testing
+
+import com.squareup.workflow.RenderContext
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.StatefulWorkflow
+import com.squareup.workflow.StatelessWorkflow
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.WorkflowAction.Companion.enterState
+import com.squareup.workflow.renderChild
+import com.squareup.workflow.stateless
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.fail
+
+class RenderTesterTest {
+
+  @Test fun `stateless input and rendering`() {
+    val workflow = Workflow.stateless<String, String, String> { input, context ->
+      return@stateless "input: $input"
+    } as StatelessWorkflow
+
+    val result = workflow.testRender("start")
+
+    assertEquals("input: start", result.rendering)
+  }
+
+  @Test fun `stateful workflow gets state`() {
+    val workflow = object : StatefulWorkflow<String, String, Nothing, String>() {
+      override fun initialState(
+        input: String,
+        snapshot: Snapshot?
+      ): String = fail("Expected initialState not to be called.")
+
+      override fun render(
+        input: String,
+        state: String,
+        context: RenderContext<String, Nothing>
+      ): String {
+        return "input=$input, state=$state"
+      }
+
+      override fun snapshotState(state: String): Snapshot =
+        fail("Expected snapshotState not to be called.")
+    }
+
+    val result = workflow.testRender(input = "foo", state = "bar")
+
+    assertEquals("input=foo, state=bar", result.rendering)
+  }
+
+  @Test fun `assert no composition`() {
+    val workflow = Workflow.stateless<Nothing, Unit> { Unit } as StatelessWorkflow
+
+    val result = workflow.testRender()
+
+    result.assertNoWorkflowsRendered()
+    result.assertNoWorkersRendered()
```

Comment:
Nit: do we really render a worker? Don't we run them?

Reply:
Spent a whole 5 minutes googling the correct tenses of "to run", I think I got it right now.